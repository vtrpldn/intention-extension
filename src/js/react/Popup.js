import React from 'react'
import ReactDOM from 'react-dom'

const Popup = () => {
  return (
    'Well, hello there, my name is Popup'
  )
}

const app = document.createElement('div')
document.body.appendChild(app)
ReactDOM.render(<Popup />, app)

export default Popup
