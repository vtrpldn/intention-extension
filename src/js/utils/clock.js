export const secondToClock = (seconds) => {
  let date = new Date(null)
  date.setSeconds(seconds)
  const timeString = date.toISOString().substr(11, 8)
  return timeString
}