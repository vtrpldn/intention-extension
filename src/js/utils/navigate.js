export const navigationTarget = {
  NEW_WINDOW: 'new-window',
  NEW_TAB: 'new-tab',
  CURRENT_TAB: 'current-tab'
}

export const navigate = (url, target = navigationTarget.NEW_TAB) => {
  switch (target) {
    case navigationTarget.NEW_WINDOW:
      return chrome.windows.create({ url, focused: true, type: 'normal' })
    case navigationTarget.CURRENT_TAB:
      return chrome.tabs.update({ url, active: true })
    default:
      return chrome.tabs.create({ url, active: true })
  }
}
