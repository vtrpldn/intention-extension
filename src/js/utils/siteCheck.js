export const getUrlListStatus = (list, url) => {
  console.log('list:', list)
  console.log('url:', url)
  if (list === []) {
    return false
  }
  return !!list.filter((v) => url.includes(v)).length
}

export const getUrlActiveStatus = (activeSites, url) => {
  console.log('activeSites:', activeSites)
  console.log('url:', url)
  if (activeSites === []) {
    return false
  }
  return !!activeSites.map(v => v.url).filter((v) => v === url).length
}
