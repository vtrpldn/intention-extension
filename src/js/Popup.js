import React from 'react'
import ReactDOM from 'react-dom'
import styled from 'styled-components'

import { getUrlListStatus } from './utils/siteCheck'
import { phrome } from './utils/wrappers/phrome'

import GlobalStyle from './GlobalStyle'
import Button from './components/Button/Button'
import Title from './components/Title/Title'

const Wrapper = styled.div`
  width: 300px;
  height: 400px;
  padding: 20px 25px;
  background: black;
  color: white;
`

class Popup extends React.Component {
  constructor () {
    super()
    this.state = {
      currentActiveUrl: '',
      isCurrentUrlListed: null,
      activeSites: []
    }
  }

  async componentDidMount () {
    const currentActiveUrl = await phrome.tabs.getCurrentUrl()
    const {siteList, activeSites} = await phrome.storage.get(['siteList', 'activeSites'])

    console.log('montou e peguei o', activeSites)
    
    this.setState({
      currentActiveUrl,
      isCurrentUrlListed: getUrlListStatus(siteList, currentActiveUrl),
      activeSites
    })
  }

  toggleCurrentToList () {
    chrome.runtime.sendMessage({
      type: 'TOGGLE_CURRENT_SITE',
      currentActiveUrl: this.state.currentActiveUrl,
      isCurrentUrlListed: this.state.isCurrentUrlListed
    }, () => {
      // Refactor this to include into state
      chrome.tabs.query({ active: true, currentWindow: true }, function (tabs) {
        chrome.tabs.update(tabs[0].id, { url: tabs[0].url })
        window.close()
      })
    })
  }

  render () {
    const {activeSites, isCurrentUrlListed} = this.state

    console.log('activeSites:', activeSites)

    return (
      <Wrapper>
        <GlobalStyle />
        {activeSites.length > 0 ? (
          <div>
            <Title text="Active timers:" />
            {activeSites.map( (val, ind) => <Title key={ind} text={val.url}  />)}
          </div> 
        ) : (
          ''
        )}   
        <Button
          block
          text={isCurrentUrlListed ? 'Remove this site from list' : 'Add this site to list'}
          onClick={() => this.toggleCurrentToList()}
        />
      </Wrapper>
    )
  }
}

const app = document.createElement('div')
document.body.appendChild(app)
ReactDOM.render(<Popup />, app)
