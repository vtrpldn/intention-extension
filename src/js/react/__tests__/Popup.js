import React from 'react'
import { shallow } from 'enzyme'
import Popup from '../Popup'
import '../../utils/enzyme'

describe('Popup component testing with Enzyme', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Popup />)
    expect(wrapper).toMatchSnapshot()
  })
})
