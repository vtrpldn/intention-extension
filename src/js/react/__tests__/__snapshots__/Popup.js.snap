// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup component testing with Enzyme renders without crashing 1`] = `ShallowWrapper {}`;
